packages/connect-agreement/subgraph/src/Agreement.ts

```diff
@@ -294,6 +270,91 @@ function updateCollateralRequirement(agreement: Address, disputable: Address, co
   requirement.save()
 }
 
+function createStakingMovement(agreement: Address, actionId: BigInt, type: string, event: ethereum.Event): void {
+  const agreementApp = AgreementContract.bind(agreement)
+  const actionData = agreementApp.getAction(actionId)
+  const collateralData = agreementApp.getCollateralRequirement(actionData.value0, actionData.value2)
+
+  const user = actionData.value4
+  const token = collateralData.value0
+  const collateralAmount = collateralData.value2
+
+  if (collateralAmount.equals(BigInt.fromI32(0))) {
+    return
+  }
+
+  const factory = StakingFactoryContract.bind(agreementApp.stakingFactory())
+  const stakingAddress = factory.getInstance(token)
+  const staking = updateStaking(stakingAddress, token, user)
+
+  const id = buildStakingMovementId(token, user, buildId(event))
+  const movement = new StakingMovement(id)
+  movement.staking = staking.id
+  movement.agreement = agreement.toHexString()
+  movement.action = buildActionId(agreement, actionId)
+  movement.createdAt = event.block.timestamp
+
+  if (type == 'new') {
+    movement.amount = collateralAmount
+    movement.actionState = 'Scheduled'
+    movement.collateralState = 'Locked'
+  } else if (type == 'challenged') {
+    movement.amount = collateralAmount
+    movement.actionState = 'Challenged'
+    movement.collateralState = 'Challenged'
+    staking.challenged = staking.challenged.plus(collateralAmount)
+  } else if (type == 'settled') {
```

Comment:
I wonder if we need to check that the `collateralAmount` be less than the `staking.challenged`?